### DISCUSSION TITLE
SPL FILE

### DISCUSSION THREAD

**Discussion by GZRGZR** [sumatrapdfreader/sumatrapdf#3972] [General]
In the old version I was able to read SPL files using the software, in the new versions I was unable to open these files, was the support removed on purpose or is there a bug [I searched the forum for information and did not find it]

**Comment by GitHubRulesOK** [comment]
There are many types of .spl 
- Those based on flash will be unlikely to be supported now in Windows as Adobe retired Shock Wave Flash as too insecure and easily exploited, MS should have removed that support code in windows 7+ ?.
- Those that are partial printouts while SPooLing will not be one fixed format, thus it depends what print driver is currently generating them .
- Several others are plain text formats
 
SumatraPDF has a file type assessment based in part on file extension and file header and "As Far As I Know" SPL is not a known extension.

Where did they originate ?

**Reply by GZRGZR** [reply]
These are files that I sent to print on an offline printer, and until I connect to the printer they are in SPL format and I need to see what is written in them. As I mentioned, in version 3.3.3 it does work, but it is a problem to open another SUMATRA [the old version] for them every time and even when I open it, it reopens all the files that are already open in the already open SUNTRA.

**Reply by GitHubRulesOK** [reply]
as I said no way to know what they are as they could be ones that need GhostScript such as PDL or PostScript or may be XPS or PDF or many others (SPL is just an alternative extension type compared to PRN) they are files designed for printers not readers so you need to check the contents for structure of header. if 3.3.3 reads them but 3.6 does not, it may be the name is masking the natural extension or even the type of file has a different structure compared to what was expected. you could zip up a small 1 page example and post into comments  here. 

To reduce the number of files opened switch off remember files in options.

Spool files are deleted on successful run so had to search Microsoft problems to find a sample (the issue there was XPS could open one but not the other since their example was an XPS printer file. Long story short Newer SumatraPDF will not load extensions without there being a known format and spooler files are an unknown however 3.3.3 detector clearly sniffs out the Zip format as similar to an XPS 3.6 needs the extension defined for which type of file to load into relevant engine.

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/fb3a6c5e-1b0c-4b21-a59b-5afadcd33663)

**Comment by kjk** [comment]
If it used to work then please open a new issue and provide a test file.

We try to detect type of the file based on the content so if it's a file that would be recognized, we should display it. Maybe the logic for detection changed.
